bigframes/core/compile/compiler.py

```diff
@@ -96,6 +96,42 @@ def compile_readlocal(node: nodes.ReadLocalNode, ordered: bool = True):
         return ordered_ir.to_unordered()
 
 
+@_compile_node.register
+def compile_cached_table(node: nodes.CachedTableNode, ordered: bool = True):
+    full_table_name = f"{node.project_id}.{node.dataset_id}.{node.table_id}"
+    hidden_col_names = [
+        col for col in node.ordering.referenced_columns if col not in node.schema.names
+    ]
+    used_columns = (
+        *node.schema.names,
+        *hidden_col_names,
+    )
```

Comment:
I wonder if it would be better to move this "used columns" logic to a more central place such as CachedTableNode itself? Seems like something that could potentially get out of sync if we add other kinds of hidden columns.

Reply:
Moved logic to define "hidden columns" to node definitions.